index.js

```diff
@@ -161,16 +161,41 @@ module.exports = class TrailsApp extends events.EventEmitter {
   }
 
   /**
+   * Extend the once emiter reader for accept multi valid events
+   */
+  onceAny (events, handler) {
+    if (!events)
+      return
+    if (!(events instanceof Array))
+      events = [events]
+
+    const cb = (e) => {
+      this.removeListener(e, cb)
+      handler.apply(this, Array.prototype.slice.call(arguments, 0))
+    }
+
+    events.forEach(e => {
+      this.addListener(e, cb)
+    })
+  }
+
+  /**
    * Resolve Promise once all events in the list have emitted
    * @return Promise
    */
   after (events) {
     if (!Array.isArray(events)) {
       events = [ events ]
     }
-
     return Promise.all(events.map(eventName => {
-      return new Promise(resolve => this.once(eventName, resolve))
+      return new Promise(resolve => {
+        if (eventName instanceof Array){
+          this.onceAny(eventName, resolve)
```

Comment:
We can replace the this.once per onceAny cos the fuctions transform the stings to array if is necessary